common/flatpak-dir.c

```diff
@@ -1153,6 +1154,57 @@ flatpak_dir_remove_oci_files (FlatpakDir   *self,
   return TRUE;
 }
 
+static gchar *
+flatpak_dir_revokefs_fuse_create_mountpoint (const gchar *ref,
+                                             GError     **error)
+{
+  g_autoptr(GFile) cache_dir = NULL;
+  g_auto(GStrv) parts = NULL;
+  g_autofree gchar *cache_dir_path = NULL;
+  g_autofree gchar *mnt_dir = NULL;
+  g_autofree gchar *mountpoint = NULL;
+
+  cache_dir = flatpak_ensure_system_user_cache_dir_location (error);
+  if (cache_dir == NULL)
+    return NULL;
+
+  parts = flatpak_decompose_ref (ref, NULL);
+  cache_dir_path = g_file_get_path (cache_dir);
+  mnt_dir = g_strdup_printf ("%s-XXXXXX", parts[1]);
+  mountpoint = g_mkdtemp_full (g_build_filename (cache_dir_path, mnt_dir, NULL), 0755);
```

Comment:
If g_mkdtemp_full returns NULL you need to call glnx_set_error_from_errno(), etc.

Reply:
handled error case as suggested